csi/volumeutils.py

```diff
@@ -610,11 +610,65 @@ def delete_volume(volname):
         volhash=vol.volhash,
         hostvol=vol.hostvol
     ))
+
     # Check for mount availability before deleting the volume
     retry_errors(os.statvfs, [os.path.join(HOSTVOL_MOUNTDIR, vol.hostvol)],
                  [ENOTCONN])
 
+    filename = vol.hostvol + ".info"
+    with open(os.path.join(VOLINFO_DIR, filename)) as info_file:
+        storage_data = json.load(info_file)
+
+    onDelete = storage_data.get("onDelete", "delete")
+
     volpath = os.path.join(HOSTVOL_MOUNTDIR, vol.hostvol, vol.volpath)
+
+    # Get brick data
+    info_file_path = os.path.join(HOSTVOL_MOUNTDIR, vol.hostvol, "info", vol.volpath)
+    with open(info_file_path + ".json", "r") as info_file:
+        brick_data = json.load(info_file)
+
+    path_prefix = brick_data["path_prefix"]
+
+    logging.info(logf(
+        "onDelete data",
+        hostvolume=vol.hostvol,
+        onDelete=onDelete,
+        volpath=volpath,
+        destination=os.path.join(HOSTVOL_MOUNTDIR, vol.hostvol, vol.volpath),
+        path_prefix=path_prefix
+    ))
+
+    if onDelete == "archive":
```

Comment:
Do you feel skipping this step https://github.com/kadalu/kadalu/blob/devel/csi/quota-crawler.sh#L22 for `archive` pvcs is good ?

Reply:
Since the data is still preserved, should the stats of Hosting Volume be updated. I think when the archived data is manually removed as in reclaimPolicy: retain, the occupied size should be subtracted.

Reply:
Ack, I see little overhead in updating the size even for archived pvcs. Above can be disregarded then.